### PR TITLE
[Translation] Improve tests coverage

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

Improves overall coverage on the Translation component.

### CODE REVIEW

File: src/Symfony/Component/Translation/Formatter/MessageFormatter.php [hunk 1]
```diff
@@ -39,11 +39,7 @@ public function __construct(TranslatorInterface $translator = null, IntlFormatte
      */
     public function format(string $message, string $locale, array $parameters = [])
     {
-        if ($this->translator instanceof TranslatorInterface) {
-            return $this->translator->trans($message, $parameters, null, $locale);
-        }
-
-        return strtr($message, $parameters);
+        return $this->translator->trans($message, $parameters, null, $locale);
```
Comment (alexandre-daubois) [inline src/Symfony/Component/Translation/Formatter/MessageFormatter.php:39]: `$this->translator` cannot be anything else than a `TranslatorInterface`, as typed in the constructor.

**Review by OskarStark — APPROVED** [review]

**Review by welcoMattic — APPROVED** [review]

**Review by fabpot — APPROVED** [review]
Merging it in 6.4 as this is not a new feature and code changes.

**Comment by fabpot** [comment]
Can you rebase on 6.4?

**Comment by fabpot** [comment]
Thank you @alexandre-daubois.

**Pull request merged**